proxy/src/main.rs

```diff
@@ -66,11 +67,16 @@ async fn main() -> anyhow::Result<()> {
     let proxy_address: SocketAddr = args.get_one::<String>("proxy").unwrap().parse()?;
     info!("Starting proxy on {proxy_address}");
     let proxy_listener = TcpListener::bind(proxy_address).await?;
+    let proxy_cancellation_token = CancellationToken::new();
 
     let mut tasks = vec![
-        tokio::spawn(handle_signals()),
+        tokio::spawn(handle_signals(proxy_cancellation_token.clone())),
         tokio::spawn(http::server::task_main(http_listener)),
-        tokio::spawn(proxy::task_main(config, proxy_listener)),
+        tokio::spawn(proxy::task_main(
+            config,
+            proxy_listener,
+            proxy_cancellation_token,
+        )),
         tokio::spawn(console::mgmt::task_main(mgmt_listener)),
     ];
 
```

Comment:
New thread for `wss` handling (one line below this).

Do they need to keep proxy alive similar to `proxy::task_main`? If so, similar handling is needed, and we also probably want to group these "not infinite tasks" separatedly from the "infinite tasks" which we can just stop polling when we decide to exit.

Reply:
I rewrote the PR to implement the simple drain functionality we discussed offline, so marking this resolved. The new version does wait for web socket connections to terminate.